src/middleware/packages/auth/mixins/auth.js

```diff
@@ -63,29 +73,112 @@ const AuthMixin = {
     async authorize(ctx) {
       const { route, req, res } = ctx.params;
       // Extract token from authorization header (do not take the Bearer part)
-      const token = req.headers.authorization && req.headers.authorization.split(' ')[1];
-      if (token) {
+      /** @type {[string, string]} */
+      const [method, token] = req.headers.authorization && req.headers.authorization.split(' ');
+
+      if (!token) {
+        ctx.meta.webId = 'anon';
+        return Promise.reject(new E.UnAuthorizedError(E.ERR_NO_TOKEN));
+      }
+
+      if (method === 'Bearer') {
         const payload = await ctx.call('auth.jwt.verifyToken', { token });
         if (payload) {
           ctx.meta.tokenPayload = payload;
           ctx.meta.webId = payload.webId;
           return Promise.resolve(payload);
         }
+      } else if (method === 'Capability') {
+        const capabilityUri = token || req.$params.capability;
+        const capability = await this.actions
+          .getValidateCapability({ capabilityUri, username: req.parsedUrl.match(/[^/]+/)[0] })
+          .catch(err => throw500(err));
+
         ctx.meta.webId = 'anon';
-        return Promise.reject(new E.UnAuthorizedError(E.ERR_INVALID_TOKEN));
+
+        req.$ctx.meta.webId = 'anon';
+        req.$ctx.meta.authorization = { capability };
+        if (capability) {
+          return Promise.resolve(null);
+        }
       }
-      ctx.meta.webId = 'anon';
-      return Promise.reject(new E.UnAuthorizedError(E.ERR_NO_TOKEN));
+
+      return Promise.reject(new E.UnAuthorizedError(E.ERR_INVALID_TOKEN));
     },
+    getValidateCapability: {
+      params: {
+        capabilityUri: {
+          type: 'string',
+          required: true
+        },
+        webId: {
+          type: 'string',
+          optional: true
+        },
+        username: {
+          type: 'string',
+          optional: true
+        }
+      },
+      /**
+       * Checks, if the provided capabilityUri is a valid URI and within the resource owner's cap container.
+       * @returns {Promise<object>} The stored capability object or undefined, if the capability is not valid.
+       */
+      async handler(ctx) {
+        let { capabilityUri, webId, username } = ctx.params;
+        /** @type {string} */
+        const baseUrlTrailing = urlJoin(this.settings.baseUrl, '/');
+        webId = webId || baseUrlTrailing + username;
+
+        if (!webId?.startsWith(baseUrlTrailing) || !capabilityUri?.startsWith(webId)) {
+          return undefined;
+        }
+
+        // Check, if capUri is a valid URI.
+        try {
+          // eslint-disable-next-line no-new
+          new URL(capabilityUri);
+        } catch {
+          return undefined;
+        }
+
+        // Check, if moleculer service 'capabilities' is available.
+        if (!ctx.broker.getLocalService('capabilities')) {
+          throw new Error('The capabilities service is not available');
+        }
```

Comment:
Since this code is part of SemApps, shouldn't the CapabilitiesService also be part of SemApps ? At least the code that is not specific to ActivityPods...

Reply:
Good point! Moved the container part to semapps auth.